scripts/utils/Git.ts

```diff
@@ -366,40 +388,39 @@ class Git {
         return mergeBaseHash;
     }
 
-    static async getChangedFiles(remote: string): Promise<string[]> {
+    /**
+     * Check if there are any uncommitted changes (both staged and unstaged).
+     *
+     * @returns true if there are uncommitted changes, false otherwise
+     */
+    static async hasUncommittedChanges(): Promise<boolean> {
+        try {
+            const {stdout} = await exec('git status --porcelain');
+            const status = stdout.trim();
+            return status.length > 0;
+        } catch (error) {
+            return false;
+        }
+    }
+
+    static async getChangedFileNames(fromRef: string, toRef = 'HEAD'): Promise<string[]> {
```

Comment:
Actually, this is a super easy 1-liner fix so that `check-changed` will catch uncommitted local tree changes.

```suggestion
    static async getChangedFileNames(fromRef: string, toRef?: string): Promise<string[]> {
```